### ISSUE TITLE
Exception: viper.silver.ast.MagicWand cannot be cast to viper.silver.ast.LocationAccess

### ISSUE THREAD

**Issue by fabiopakk** [viperproject/carbon#364]
Carbon is crashing with the message above when verifying the following code:

```
field val: Int

define wandA(a) acc(a.val) --* acc(a.val)

function reqA(a: Ref): Int
  requires wandA(a)

method test_wand_framing_2(a: Ref) {
  inhale acc(a.val) && a.val == 77
  package wandA(a)
}
```

This is possibly related with issue viperproject/silver#456.

**Comment by maurobringolf** [comment]
I think this is related or a duplicate of https://github.com/viperproject/carbon/issues/244.

**Comment by fabiopakk** [comment]
I've fixed this issue by implementing a consistency check which prevents programs with Magic Wands in function preconditions and predicates to be verified (both backends). I also filed viperproject/silver#478 formalizing the need to fully support Magic Wands on both backends. I've added instructions on how to revert the changes in the regression tests that were introduced by this temporary consistency check.
